### DISCUSSION TITLE
Import pywebview?

### DISCUSSION THREAD

**Discussion by Proton-V** [exodrifter/unity-python#32] [Q&A]
How i can import pywebview library?

**Comment by exodrifter** [comment]
I'm not sure if it will work, but if you add the path to the `pywebview` library like in the [PythonLibrary.cs](https://github.com/exodrifter/unity-python/blob/6d9a5908dc976ac828230d461fb6e262360ce0fb/Examples/PythonLibrary.cs) example, that's how you would do it.

**Comment by Proton-V** [comment]
How i can import(ex. urllib3)? Its will be work after build?

**Comment by Proton-V** [comment]
if i us var paths = engine.GetSearchPaths();
paths.Add(Application.dataPath + "/Python/Lib");
engine.SetSearchPaths (paths); i have error SyntaxErrorException: unexpected token ',' on 			source.Execute(scope);

**Comment by exodrifter** [comment]
@Proton-V Can you post the C# code and the python script you are executing? It sounds like the error is occurring on a line you omitted from your example.

**Comment by Proton-V** [comment]
now error:
SyntaxErrorException: unexpected indent
Microsoft.Scripting.ErrorSink.Add (Microsoft.Scripting.SourceUnit source, System.String message, Microsoft.Scripting.SourceSpan span, System.Int32 errorCode, Microsoft.Scripting.Severity severity) (at <96bae61c64314af5927380c0b6bd16cf>:0)
IronPython.Compiler.Parser.ReportSyntaxError (System.Int32 start, System.Int32 end, System.String message, System.Int32 errorCode) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseTestListAsExprError () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseTestListAsExpr () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseExprStmt () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseSmallStmt () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseSimpleStmt () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseStmt () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseFileWorker (System.Boolean makeModule, System.Boolean returnValue) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.Parser.ParseFile (System.Boolean makeModule, System.Boolean returnValue) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.ParseAndBindAst (Microsoft.Scripting.Runtime.CompilerContext context) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.CompilePythonCode (Microsoft.Scripting.SourceUnit sourceUnit, Microsoft.Scripting.CompilerOptions options, Microsoft.Scripting.ErrorSink errorSink) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.CompileSourceCode (Microsoft.Scripting.SourceUnit sourceUnit, Microsoft.Scripting.CompilerOptions options, Microsoft.Scripting.ErrorSink errorSink) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.SourceUnit.Compile (Microsoft.Scripting.CompilerOptions options, Microsoft.Scripting.ErrorSink errorSink) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope, Microsoft.Scripting.ErrorSink errorSink) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.Hosting.ScriptSource.Execute (Microsoft.Scripting.Hosting.ScriptScope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Exodrifter.UnityPython.Examples.PythonHelloWorld.Start () (at Assets/Python/Examples/PythonHelloWorld.cs:28

namespace Exodrifter.UnityPython.Examples
{
	public class PythonHelloWorld : MonoBehaviour
	{
		void Start()
		{
			var engine = Python.CreateEngine();
			var scope = engine.CreateScope();

			var libs = new[] {
	"C:\\Program Files\\IronPython 2.7\\Lib",
	"C:\\Program Files\\IronPython 2.7\\DLLs",
	"C:\\Program Files\\IronPython 2.7",
	"C:\\Program Files\\IronPython 2.7\\lib\\site-packages"
};

			engine.SetSearchPaths(libs);

			string code = "import urllib.request\n import json\n";
			code += "url = urllib.request.urlopen('http://maps.googleapis.com/maps/api/geocode/json?address=google').read()\n";
			code += "json = json.loads(url.decode())";


			var source = engine.CreateScriptSourceFromString(code);
			source.Execute(scope);

			Debug.Log(scope.GetVariable<string>("json"));
		}
	}
}

**Comment by Proton-V** [comment]
Another error:
TypeErrorException: not expecting type '<type 'bytes'>'
Microsoft.Scripting.Runtime.LightExceptions.ThrowException (Microsoft.Scripting.Runtime.LightExceptions+LightException lightEx) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Runtime.LightExceptions.CheckAndThrow (System.Object value) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`2[T0,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.HandleException (Microsoft.Scripting.Interpreter.InterpretedFrame frame, System.Exception exception) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet] (T0 arg0, T1 arg1) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonCallTargets.OriginalCallTarget1 (IronPython.Runtime.PythonFunction function, System.Object arg0) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.FunctionCaller`1[T0].Call1 (System.Runtime.CompilerServices.CallSite site, IronPython.Runtime.CodeContext context, System.Object func, T0 arg0) (at <264a665625104e3c9b8ba1905cf1b515>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`6[T0,T1,T2,T3,T4,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run4[T0,T1,T2,T3,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.Ast.CallExpression+Invoke1Instruction.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run6[T0,T1,T2,T3,T4,T5,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonCallTargets.OriginalCallTarget5 (IronPython.Runtime.PythonFunction function, System.Object arg0, System.Object arg1, System.Object arg2, System.Object arg3, System.Object arg4) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`8[T0,T1,T2,T3,T4,T5,T6,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run7[T0,T1,T2,T3,T4,T5,T6,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute6[T0,T1,T2,T3,T4,T5,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.DynamicInstruction`7[T0,T1,T2,T3,T4,T5,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run6[T0,T1,T2,T3,T4,T5,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute5[T0,T1,T2,T3,T4,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.DynamicInstruction`6[T0,T1,T2,T3,T4,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run7[T0,T1,T2,T3,T4,T5,T6,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonCallTargets.OriginalCallTarget6 (IronPython.Runtime.PythonFunction function, System.Object arg0, System.Object arg1, System.Object arg2, System.Object arg3, System.Object arg4, System.Object arg5) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`9[T0,T1,T2,T3,T4,T5,T6,T7,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run9[T0,T1,T2,T3,T4,T5,T6,T7,T8,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7, T8 arg8) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute8[T0,T1,T2,T3,T4,T5,T6,T7,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.DynamicInstruction`9[T0,T1,T2,T3,T4,T5,T6,T7,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run8[T0,T1,T2,T3,T4,T5,T6,T7,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute7[T0,T1,T2,T3,T4,T5,T6,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.DynamicInstruction`8[T0,T1,T2,T3,T4,T5,T6,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run7[T0,T1,T2,T3,T4,T5,T6,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonCallTargets.OriginalCallTarget6 (IronPython.Runtime.PythonFunction function, System.Object arg0, System.Object arg1, System.Object arg2, System.Object arg3, System.Object arg4, System.Object arg5) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`9[T0,T1,T2,T3,T4,T5,T6,T7,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run6[T0,T1,T2,T3,T4,T5,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute5[T0,T1,T2,T3,T4,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
IronPython.Runtime.FunctionCaller`3[T0,T1,T2].Call3 (System.Runtime.CompilerServices.CallSite site, IronPython.Runtime.CodeContext context, System.Object func, T0 arg0, T1 arg1, T2 arg2) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Method+MethodBinding`2[T0,T1].SelfTarget (System.Runtime.CompilerServices.CallSite site, IronPython.Runtime.CodeContext context, System.Object target, T0 arg0, T1 arg1) (at <264a665625104e3c9b8ba1905cf1b515>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Microsoft.Scripting.Interpreter.DynamicInstruction`5[T0,T1,T2,T3,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet] (T0 arg0, T1 arg1) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonScriptCode.RunWorker (IronPython.Runtime.CodeContext ctx) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.PythonScriptCode.Run (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.RuntimeScriptCode.InvokeTarget (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.RuntimeScriptCode.Run (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope, Microsoft.Scripting.ErrorSink errorSink) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.Hosting.ScriptSource.Execute (Microsoft.Scripting.Hosting.ScriptScope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Exodrifter.UnityPython.Examples.PythonHelloWorld.Start () (at Assets/Python/Examples/PythonHelloWorld.cs:36)


code:
namespace Exodrifter.UnityPython.Examples
{
	public class PythonHelloWorld : MonoBehaviour
	{
		void Start()
		{
			var engine = Python.CreateEngine();
			var scope = engine.CreateScope();

			var libs = new[] {
	"C:\\Program Files\\IronPython 2.7\\Lib",
	"C:\\Program Files\\IronPython 2.7\\DLLs",
	"C:\\Program Files\\IronPython 2.7",
	"C:\\Program Files\\IronPython 2.7\\lib\\site-packages"
};

			engine.SetSearchPaths(libs);

            string code = "import urllib3\n";
			code += "import json\n";


			code += "http = urllib3.PoolManager()\n";
			code += "url = http.request('GET','http://maps.googleapis.com/maps/api/geocode/json?address=google').data\n";
			code += "json = json.loads(url.decode())\n";


			var source = engine.CreateScriptSourceFromString(code);
			source.Execute(scope);

			//Debug.Log(scope.GetVariable<string>("json"));
		}
	}
}

**Comment by Proton-V** [comment]
and if i write "import urllib.request"
i get error: ImportException: No module named request

**Comment by exodrifter** [comment]
`import urllib.request` will not work, because the `request` module doesn't exist in Python 2.x; it only exists in Python 3.x. This [stack overflow](https://stackoverflow.com/questions/3745771/urllib-request-in-python-2-7) question suggests that the equivalent Python 2.x code `urllib2.urlopen`.

As for your other error about `not expecting type '<type 'bytes'>'`, I'm not really sure what might be causing that problem. Maybe you're passing in the wrong type somewhere in your Python code? This error might also be caused by the fact that you're trying to run Python 3.x code.
